controllers/front/ProductController.php

```diff
@@ -106,11 +106,17 @@ public function init()
 
         $id_product = (int) Tools::getValue('id_product');
 
+        // Get product by EAN-13
+        $ean13_product = Tools::getValue('ean13_product');
+        if ($ean13_product !== null) {
+            $id_product = Db::getInstance()->getValue('SELECT `id_product` FROM `'._DB_PREFIX_.'product` p WHERE p.`ean13` = '. (int) $ean13_product);
+            $id_product = (!$id_product) ? null : $id_product;
+        }
```

Comment:
```suggestion
        $ean13_product = Tools::getValue('ean13_product');
        if ($ean13_product !== null && $id_product !== 0) {
            $id_product = Db::getInstance()->getValue('SELECT `id_product` FROM `'._DB_PREFIX_.'product` p WHERE p.`ean13` = '. (int) $ean13_product);
            $id_product = $id_product ?? null;
        }
```
Do not execute SQL request if you already have an id_product?
And simplify the ternary check :) 
But LGTM otherwise

Reply:
I agree :-)